### ISSUE TITLE
Improve roles

### ISSUE THREAD

**Issue by KaterynaBasik** [equinor/flyt#740]
Currently there are following roles in Flyt

- Owner (Default role for the one who creates the process initially, can share write access to others)
- Admin (can share write access to others)
- Contributor

**User input 1 (by Fredrik Liknes, feedback ID 235 ):**
As a contributor I would like to be able to share write access to a VSM without requesting the owner to add new contributors.

**Comment by nesadrian** [comment]
We have previously discussed if we should remove owner altogether and only have admin and contributor roles since owner and admin has the same privileges anyways. The idea of everyone having the same role has also been brought up before. This would solve #247 and since Flyt is limited to Equinor users anyways, removing these accesses might not be a big concern.

**Comment by askbulle** [comment]
All three roles should be combined into role Write and that role should be able to both write and share write access

**Comment by askbulle** [comment]
If Quick fix priority should be high. If not, re-assess prio

**Comment by srvEq** [comment]
Hi @askbulle 
As discussed three roles(Owner, Admin, Contributor) will combine into "Collaborator" role will have the both write and share write access.

Does "Collaborator" role also have the privilege to delete the Process?

**Comment by askbulle** [comment]
Yes, Collaborator should have all access that Owner and Admin has. 

Can we have a look at the message ? It seems that when some text is bolded more people actually see it. 

![Image](https://github.com/user-attachments/assets/01641e36-b3ff-457e-b889-9338c4cdcb0b)

**Comment by askbulle** [comment]
Suggested text:

Are you sure you want to delete this process? By doing so you will delete **ALL VERSIONS** as they will not be recoverable.

**Comment by srvEq** [comment]
Changes deployed in Test env.
@AnjaliPandit20 @SIJAIEQUINOR @rvaidya3007

**Comment by srvEq** [comment]
Issue1, Issue3 both happens because of existing data where it has roles as admin, owner, contributor. It will work fine once existing data is handled.

My suggestion is to do the data cleanse which refers to change all the existing roles in DB to contributor, so there won't be any data loss.

The above issues has to be addressed from Backend. 
@sarsi0801 Can you please review the above issue.

Issue 2
Thanks for the finding, When only one user has access to a process then we can disable/restrict the user from providing the option to remove.Does this sound good ? 

@SIVN1 @luan-droid @ThomasBracho1987 @jsundarn

**Comment by sarsi0801** [comment]
@srvEq @SIVN1  I will change the existing roles in DB to contributor for few old process then i will test above issue, if it works fien then i will update roles to contributor for all old process. This will resolve Issue 1 and 3.

**Comment by SIVN1** [comment]
@srvEq  We can go with the solution which you have provided for Issue 2. Thank you

CC: @AnjaliPandit20 @luan-droid @ThomasBracho1987 @jsundarn @rvaidya3007

**Comment by sarsi0801** [comment]
@srvEq 
I have updated the role for all old process to contributor. 
You can test the Issue1 and Issue3.

**Comment by luan-droid** [comment]
Issue 1: I can now delete my old processes, it works fine.

**Comment by srvEq** [comment]
Issue 2 is fixed and deployed in test env.
@SIVN1 @luan-droid @ThomasBracho1987

**Comment by SIVN1** [comment]
@srvEq  @sarsi0801 Verified issue 2 and adding same name again issue in Test. It works fine


<img width="954" alt="Image" src="https://github.com/user-attachments/assets/98127812-d6e1-434b-b85b-25039de3e8b3" />

<img width="959" alt="Image" src="https://github.com/user-attachments/assets/390a6e8a-2fc3-4b4c-a8b4-2006ebcc093e" />

CC: @AnjaliPandit20 @ThomasBracho1987 @luan-droid @SIJAIEQUINOR @rvaidya3007

**Comment by srvEq** [comment]
Changes deployed in QA. @sarsi0801 is looking into DB changes.

**Comment by sarsi0801** [comment]
DB changes are done in QA environment.

**Comment by srvEq** [comment]
Ready to test the changes in QA environment.
@SIVN1 @luan-droid @ThomasBracho1987

**Comment by luan-droid** [comment]
It works fine, but the pop up window for renaming looks weird 

![Image](https://github.com/user-attachments/assets/93511ced-c4d7-4939-96a0-0604bed9046c)

**Comment by srvEq** [comment]
The above issue will be tracked in #884. Hence closing this.
@luan-droid @SIVN1 @ThomasBracho1987 
CC: @AnjaliPandit20 @SIJAIEQUINOR @rvaidya3007

**Comment by srvEq** [comment]
Changes are ready to test in both Test and QA env.
@SIVN1 @luan-droid @ThomasBracho1987

**Comment by luan-droid** [comment]
Tested in QA and Test, works fine. 

- _All three roles should be combined into role "Collaborator" and that role should be able to both write and share write access_


![Image](https://github.com/user-attachments/assets/3efc482b-af3a-4659-873d-d7db7d8a4beb)

![Image](https://github.com/user-attachments/assets/dab060c4-4abb-49eb-a742-53bb76de3aae)

**Comment by AnjaliPandit20** [comment]
@sarsi0801 : List all processes with owner, admin name in Production database and check if the user is still with Equinor?
check on the process, if it is minimal, agree on deletion 
for reference: 1586

Short id : THOKN

**Comment by SIVN1** [comment]
Verified in Prod, Can contributor able to write and remove one of the contributor from the process.

<img width="953" alt="Image" src="https://github.com/user-attachments/assets/5da252ae-b36c-4a75-ad01-743bf1ef5ab3" />

**Known issue:**

If we click "+" button multiple times in adding the contributor. Respective contributor name is showing multiple times in the list

<img width="957" alt="Image" src="https://github.com/user-attachments/assets/2b5e82f7-d29b-4402-8a27-7dc7b7b77f9f" />

CC: @AnjaliPandit20 @luan-droid @srvEq @sarsi0801 @MEPA2024
